pm/recipient.go

```diff
@@ -86,7 +89,7 @@ type recipient struct {
 	maxfacevalue *big.Int
 
 	senderNonces map[string]*struct {
-		nonce           uint32
+		nonceSeen       map[uint32]byte
```

Comment:
Is there an advantage of using `byte` here instead of a `bool`?

IIUC a `bool` should be 1 byte which would be the same size as a `byte` [1] so I don't think there are any storage benefits of using `byte`. If this is the case, I suggest using a `bool` as I think using true/false values will be clearer for code readability than using 0/1.

[1] Can verify using `unsafe.Sizeof()` on a `bool` var and a `byte` var.